main.py

```diff
@@ -100,7 +98,11 @@ def tracepath(self, entries, stored_flows):
             if 'dl_vlan' in entries:
                 trace_step['in'].update({'vlan': entries['dl_vlan'][-1]})
             switch = self.controller.get_switch_by_dpid(entries['dpid'])
-            result = self.trace_step(switch, entries, stored_flows)
+            if (switch is None) or \
+                    (switch.dpid not in list(stored_flows.keys())):
```

Comment:
`switch.dpid not in list` is O(n) average time complexity, this can be done in O(1) with `switch.dpid not in stored_flows`, could you replace this accordingly?

Reply:
@viniarck, sorry. Of course, that's right, that sentence stayed like this because I did tests with the stored_flow structure and forgot to change it at the end. So sorry.

Reply:
You are right.